python/cuml/dask/common/base.py

```diff
@@ -147,12 +235,14 @@ def _run_parallel_func(self,
         -------
         y : dask cuDF (n_rows, 1)
         """
+        X_d = X.to_delayed()
+
         if output_collection_type is None:
             output_collection_type = self.datatype
 
-        X_d = X.to_delayed()
-
-        model = dask.delayed(self.local_model, pure=True, traverse=False)
+        model = dask.delayed(self._get_internal_model(),
```

Comment:
I think it's helpful in naming when we use `model_delayed` or `model_d` to indicate that this is a delayed objcet.